subprojects/core/src/main/java/org/gradle/internal/service/scopes/WorkerSharedProjectScopeServices.java

```diff
@@ -159,6 +162,11 @@ protected ProjectScopedCacheBuilderFactory createProjectScopedCache(TemporaryFil
     }
 
     protected DecompressionCacheFactory createDecompressionCacheFactory(TemporaryFileProvider temporaryFileProvider, UnscopedCacheBuilderFactory unscopedCacheBuilderFactory) {
-        return new DefaultDecompressionCacheFactory(() -> new DefaultProjectScopedCacheBuilderFactory(temporaryFileProvider.newTemporaryFile(".cache"), unscopedCacheBuilderFactory));
+        return new DefaultDecompressionCacheFactory(() -> new DefaultProjectScopedCacheBuilderFactory(temporaryFileProvider.newTemporaryFile(".cache"), unscopedCacheBuilderFactory), buildExpansionCacheDir());
+    }
+
+    private File buildExpansionCacheDir() {
+        File rootExpansionCacheDir = new File(projectCacheDir, DecompressionCache.EXPANSION_CACHE_KEY);
+        return new File(rootExpansionCacheDir, projectDir.getName());
```

Comment:
🤔 What happens if this is a shared directory? I think that implies that only one subproject can extract a zip at a time?

The projectDir.name isn't unique enough.  You could easily have two "api" subprojects and they would overlap here. 

Would it be good enough if this cache was shared across the entire build?

Reply:
I was thinking the increased contention between projects with the same name wasn't worth worrying about.

But if we want to avoid this, we can use the project path, which should be unique.  I've done this here: https://github.com/gradle/gradle/pull/27005/commits/91a7ddd4fe83657bd5e5bffb7e24ed58ee8c9373.  There is a new test in `ConcurrentArchiveIntegrationTest` to demo a build with `/lib` and `/utils/lib` using unique cache content dirs `_lib` and `_utils_lib`.

Let me know WYT about this approach, and how the root project dir is supplied in order to make this calculation.

Reply:
It's expected that tests here could be broke (other than the tests relating to this cache), due to the new required parameter.  I'll wait until after we decide whether or not to proceed with this to fix them.